impl/client/src/main/javascript/web/pentaho/visual/scene/Base.js

```diff
@@ -102,11 +120,30 @@ define([
     },
 
     /**
-     * Marks this node's `$root` "variable" with `this` as a value.
+     * Gets the associated visualization view.
+     *
+     * @type {!pentaho.visual.base.View}
+     * @readOnly
+     */
+    get view() {
+      return this.root.__view;
+    },
+
+    /**
+     * Appends a given child scene in the local children array, creating one if this is the first child.
+     *
+     * @param {!pentaho.visual.scene.Base} child - The child scene.
+     * @return {number} The index at which the given child scene was inserted.
      * @private
      */
-    __setAsRoot: function() {
-      O.setConst(this.vars, "$root", this);
+    __appendChildCore: function(child) {
+
+      var children = O.getOwn(this, "children") || (this.children = []);
```

Comment:
`O.getOwn` only returns local - i.e. _own_ - properties, not inherited ones. Thus, until the array is created locally, the function will return `undefined`. The prototype's `Object.freeze([])` exists only to avoid allocating arrays for each leaf scene, while still providing the convenience to "consumers" to not have to test for `null` children before iteration, in generic code.